### PR TITLE
Feature/misc changes

### PR DESCRIPTION
**Pull request by skwerlman**
a collection of miscellaneous changes i had meant to push at various points but never got around to.

- ignore data folders
- update to distillery 2 (not used yet, i know)
- merge the various migrations into a single file; we dont need new migrations until we've actually deployed
- remove useless code/tests from database
- move crawler from gen_stage to flow
- update all dependencies to their newest allowed versions
- cache the _build dir in CI to make things faster

### CODE REVIEW

**Comment by skwerlman** [comment]
literally just remembered about this PR;  whoops

gonna update deps rq, then its just waiting on review

**Review by iggi42 — CHANGES_REQUESTED** [review]
code looks good to me.
but the builds are failing https://travis-ci.com/Belethors-General-Mods/belethor/jobs/155025830#L709
I don't have much  exp with travis, but this looks like it should be fixed before merging?

File: apps/database/lib/database/schema/mod.ex
Comment (iggi42) [inline apps/database/lib/database/schema/mod.ex]: dialyzer complains about no local return here

Comment (iggi42) [reply apps/database/lib/database/schema/mod.ex]: Database.Schema.Mod.changeset(_any(), :empty)

will never return since it differs in arguments with
positions 2 from the success typing arguments:

(
  {map(), map()}
  | %{:__struct__ => atom() | %{:__changeset__ => map(), _ => _}, atom() => _},
  :invalid | %{:__struct__ => none(), (atom() | binary()) => _}
)

File: apps/database/lib/database/schema/mod_file.ex
Comment (iggi42) [inline apps/database/lib/database/schema/mod_file.ex]: same dialyzer problem as with mod.ex here.
________________________________________________________________________________
apps/database/lib/database/schema/mod_file.ex:17:no_return
Function changeset/1 has no local return.
________________________________________________________________________________
apps/database/lib/database/schema/mod_file.ex:17:call
The call:
Database.Schema.ModFile.changeset(_any(), :empty)

will never return since it differs in arguments with
positions 2 from the success typing arguments:

(
  {map(), map()}
  | %{:__struct__ => atom() | %{:__changeset__ => map(), _ => _}, atom() => _},
  :invalid | %{:__struct__ => none(), (atom() | binary()) => _}
)
________________________________________________________________________________

File: mix.lock
Comment (iggi42) [inline mix.lock:38]: `mix deps.get` warns about phoenix_ecto 3.5.0:

 phoenix_ecto 3.5.0 RETIRED!
    (invalid)

**Review by iggi42 — CHANGES_REQUESTED** [review]
do you have time to fix these?
Otherwise I could try.

File: apps/crawler/mix.exs
Comment (iggi42) [inline apps/crawler/mix.exs:33]: we don't need flow anymore, right?
my TaskManager does the rate limiting.

File: apps/crawler/lib/crawler/site.ex
Comment (iggi42) [inline apps/crawler/lib/crawler/site.ex:1]: this is roughly your equivalent to Crawler.Provider, right?

File: apps/database/lib/database.ex
Comment (iggi42) [inline apps/database/lib/database.ex:23]: modfile, tag, image, mod types need to be moved to common, once it arrives.

Comment (iggi42) [reply apps/database/lib/database.ex:23]: then we can convert them to structs, to make type errors happen during compiling.

**Review by iggi42 — COMMENTED** [review]
ebert's issues should be fixed too

File: apps/database/config/config.exs
Comment (skwerlman) [inline apps/database/config/config.exs:42]: keep

File: apps/database/priv/repo/migrations/20180723104539_create_database.exs
Comment (skwerlman) [inline apps/database/priv/repo/migrations/20180723104539_create_database.exs:1]: keep (+ related deletions in other files)

File: etc/nginx/belethor.conf
Comment (skwerlman) [inline etc/nginx/belethor.conf:1]: we _can_ keep this; not 100% necessary tho

File: mix.exs
Comment (skwerlman) [inline mix.exs:12]: keep

File: mix.exs
Comment (skwerlman) [inline mix.exs:20]: keep

File: mix.exs
Comment (skwerlman) [inline mix.exs:103]: keep (excluding the `run` bit of setup)

**Pull request closed without merging**
